File: src/main/java/com/lcaohoanq/nocket/domain/wallet/Wallet.java
package com.lcaohoanq.nocket.domain.wallet;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.lcaohoanq.nocket.domain.user.User;
import com.lcaohoanq.nocket.base.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "wallets")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
public class Wallet extends BaseEntity {
    
    @Column(name = "balance", nullable = false)
    private Float balance;

    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    @JsonBackReference // Prevent infinite loop when serializing
    private User user;

}

File: src/main/kotlin/com/lcaohoanq/nocket/domain/wallet/Wallet.kt
package com.lcaohoanq.nocket.domain.wallet

import com.fasterxml.jackson.annotation.JsonBackReference
import com.lcaohoanq.nocket.base.entity.BaseEntity
import com.lcaohoanq.nocket.domain.user.User
import jakarta.persistence.*
import org.hibernate.envers.Audited

@Audited
@Entity
@Table(name = "wallets")
class Wallet : BaseEntity() {
    @JvmField
    @Column(name = "balance", nullable = false)
    var balance: Float? = null

    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id", nullable = false)
    @JsonBackReference // Prevent infinite loop when serializing
    var user: User? = null
}
